### PR TITLE
fix: Updated to use ethers-rs

### PR DESCRIPTION
**Pull request by Maikol**
### Removed dependencies:
```
eip-712-derive
solidity-bindgen
web3
```

### Added dependencies:
```
ethers
url
```

### Config changes
Removed the `ORACLE_CONTRACTS` env variable. This was used to get the network name, chain id, contracts addresses and RPC url. Value was split with a `:` so you had to pass `sepolia:RPC_URL`. Replaced this with two variables `REWARDS_MANAGER_CONTRACT ` and `RPC_URL`. We only really need the rewards manager address so by doing this change we can remove all contract configs and if we need to run on a different network we just need to pass the contract address and url, no need for code changes. We are also getting the chain id from the provider so we removed `common/src/chain_id.rs` since we no longer need it.

### Contract interactions
By switching from `solidity-bindgen` to `ethers-rs` we can get rid of over 8k lines in `common/src/contracts/abis.rs` and only import `availability-oracle/src/abi/RewardsManager.abi.json` which is what we want to interact with. We're also simulating the transaction to check of errors and if we detect one we log the revert reason.

### CODE REVIEW

File: availability-oracle/src/main.rs [hunk 1]
```diff
@@ -132,6 +120,16 @@ struct Config {
         help = "a comma separated list of the supported data source kinds"
     )]
     supported_data_source_kinds: Vec<String>,
+
+    #[structopt(
+        long,
+        env = "REWARDS_MANAGER_CONTRACT",
+        help = "The address of the rewards manager contract"
+    )]
+    pub rewards_manager_contract: String,
```
Comment (Theodus) [inline availability-oracle/src/main.rs:129]: Might be convenient to parse this into an `Address` upfront.
```suggestion
    pub rewards_manager_contract: Address,
```

File: common/Cargo.toml [hunk 2]
```diff
@@ -9,16 +9,13 @@ publish = false
...
 slog = { version = "2.5.2", features = ["release_max_level_trace", "max_level_trace"] }
 slog-term = "2.6.0"
 slog-async = "2.5.0"
 futures = { version="0.3.5", features=["compat"] }
 warp = "0.3.1"
 prometheus = "0.12.0"
-tokio = { version="1.10", features=["sync", "time"] }
-web3 = "0.16.0"
-solidity-bindgen = "0.1.0"
+tokio = { version="1.10", features=["sync", "time", "rt-multi-thread"] }
```
Comment (Theodus) [inline common/Cargo.toml:18]: This can be a separate PR: probably a good idea to upgrade these dependencies. I typically check for the latest versions with [cargo-outdated](https://github.com/kbknapp/cargo-outdated) (`cargo outdated -R`)

**Review by Theodus — APPROVED** [review]
LGTM. Just some suggestions

File: availability-oracle/src/main.rs (hunk 1 above)
Comment (Maikol) [reply availability-oracle/src/main.rs]: Nice, updated!

File: common/Cargo.toml (hunk 2 above)
Comment (Maikol) [reply common/Cargo.toml:18]: Definitely, I'll do it as a new PR.

Screenshot 2024-02-08 at 14 51 40
